src/module_writer.rs

```diff
@@ -759,6 +759,58 @@ pub fn write_bin(
     Ok(())
 }
 
+/// Adds a wrapper script that start the wasm binary through wasmtime.
+///
+/// Note that the wasm binary needs to be written separately by [write_bin]
+pub fn write_wasm_launcher(
+    writer: &mut impl ModuleWriter,
+    metadata: &Metadata21,
+    bin_name: &str,
+) -> Result<()> {
+    let entrypoint_script = format!(
+        r#"from pathlib import Path
+
+from wasmtime import Store, Module, Engine, WasiConfig, Linker
+
+import sysconfig
+
+def main():
+    # The actual executable
+    program_location = Path(sysconfig.get_path("scripts")).joinpath("{}")
+    # wasmtime-py boilerplate
+    engine = Engine()
+    store = Store(engine)
+    # TODO: is there an option to just get the default of the wasmtime cli here?
+    wasi = WasiConfig()
+    wasi.inherit_argv()
+    wasi.inherit_env()
+    wasi.inherit_stdout()
+    wasi.inherit_stderr()
+    wasi.inherit_stdin()
+    store.set_wasi(wasi)
+    linker = Linker(engine)
+    linker.define_wasi()
+    module = Module.from_file(store.engine, str(program_location))
+    linking1 = linker.instantiate(store, module)
+    # TODO: this is taken from https://docs.wasmtime.dev/api/wasmtime/struct.Linker.html#method.get_default
+    #       is this always correct?
+    start = linking1.exports(store).get("") or linking1.exports(store)["_start"]
+    start(store)
```

Comment:
I remember that wasmtime execution is sandboxed, so you might need to give access to file systems explicitly somehow?

Reply:
i think so, let's see if i get a response in https://github.com/bytecodealliance/wasmtime-py/issues/94 and otherwise i'll have to collect some real world use cases compatible with wasm